IBAnimatable/AnimationChainable.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  Created by jason akakpo on 01/01/2017.
+//  Copyright © 2017 IBAnimatable. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+typealias AnimationTuple = (type: AnimationType, configuration: AnimationConfiguration)
+
+struct AnimationConfiguration {
+  let damping: CGFloat
+  let velocity: CGFloat
+  let duration: TimeInterval
+  let delay: TimeInterval
+  let force: CGFloat
+
+}
+
+public class AnimationPromise<T: UIView> where T: Animatable {
+
+  private var view: T
+  private var animationList = [AnimationTuple]()
+  private var delayForNextAnimation = 0.0
+  private var completion: AnimatableCompletion?
+
+  init(view: T) {
+    self.view = view
+  }
+
+   func animationCompleted() {
+      animationList.remove(at: 0)
```

Comment:
May use `animationList.removeFirst` to make it more specify?

I got an idea here. Because `removeFirst` from an `Array` is an `O(N)` operation. It means anytime run an animation will be `O(N^2)` which is `O(N) + O(N-1) + ... + O(1)`. If we have an index for current running animations in `animationList` and then we increase the index every time. And get the `currentAnimation` via  `if let currentAnimation = animationList[currentAnimationIndex]` in line 33. 

But since we don't have a big list of animations. Currently, implementation is totally fine.